convert2rhel/actions/pre_ponr_changes/backup_system.py

```diff
@@ -77,8 +80,57 @@ def run(self):
 
         super(BackupRepository, self).run()
 
-        repo.backup_yum_repos()
-        repo.backup_varsdir()
+        self.backup_yum_repos()
+
+    def backup_yum_repos(self):
+        """Backup .repo files in /etc/yum.repos.d/ so the repositories can be restored on rollback."""
+        loggerinst.info("Backing up .repo files from %s." % DEFAULT_YUM_REPOFILE_DIR)
+
+        repo_files_backed_up = False
+
+        for repo in os.listdir(DEFAULT_YUM_REPOFILE_DIR):
+            if repo.endswith(".repo") and repo != "redhat.repo":
+                repo_path = os.path.join(DEFAULT_YUM_REPOFILE_DIR, repo)
+                restorable_file = RestorableFile(repo_path)
+                backup.backup_control.push(restorable_file)
+                repo_files_backed_up = True
+
+        if not repo_files_backed_up:
+            loggerinst.info("No .repo files backed up.")
+
+
+class BackupVariables(actions.Action):
+    id = "BACKUP_VARIABLES"
+
+    def run(self):
+        """Backup varsdir folder in /etc/{yum,dnf}/vars so the variables can be restored on rollback."""
+        loggerinst.task("Prepare: Backup variables")
+
+        super(BackupVariables, self).run()
+
+        loggerinst.info("Backing up variables files from %s." % DEFAULT_YUM_VARS_DIR)
+        self._backup_variables(path=DEFAULT_YUM_VARS_DIR)
+
+        if system_info.version.major >= 8:
+            loggerinst.info("Backing up variables files from %s." % DEFAULT_DNF_VARS_DIR)
+            self._backup_variables(path=DEFAULT_DNF_VARS_DIR)
+
+    def _backup_variables(self, path):
+        """Helper internal function to backup the variables.
+
+        :parameter path: The path for the original variable.
+        :type path: str
+        """
+        variable_files_backed_up = False
+
+        for variable in os.listdir(path):
+            variable_path = os.path.join(path, variable)
+            restorable_file = RestorableFile(variable_path)
+            backup.backup_control.push(restorable_file)
+            variable_files_backed_up = True
+
+        if not variable_files_backed_up:
+            loggerinst.info("No variables files backed up.")
```

Comment:
Not a necessary change, but we could move this to be outside the class scope. Since this is not using the `self` reference, we tend to imagine those functions as being static (i.e: not needing any internal reference to be accessed/used, so, this could be a private function that is tied to the module, rather than the class itself. 

Of course, it has the single usage of backing up variables, but if you want to make the class less polluted without things that are not really necessary there, you could move this function to be outside the class scope.

Reply:
@r0x0d If I can choose, I would rather have it there, as it's related only to this class and I have better feeling of having it together.

But your idea also makes a sense to me and I spend a bit of time thinking what I like more :D The result of having cleaner classes and having the related function in the module as private functions is attractive.

Reply:
Sounds good to me